### ISSUE TITLE
Cursive incorrectly wraps string and regex literals

### ISSUE THREAD

**Issue by cursive-ide** [cursive-ide/cursive#1770]
Reported by @cemerick on Twitter.

When *Settings->Editor->Code Style->Wrap on typing* is enabled, IntelliJ automatically wraps elements when they reach the margin. This causes String and regex literals to be broken.

![wrapping](https://cloud.githubusercontent.com/assets/5589171/26563759/55243fc0-452b-11e7-89f1-5ac6e7f04319.gif)

**Comment by cemerick** [comment]
This is true for doubles, too; for `0.3`, intellij will wrap `.3` onto its own line.

**Comment by cemerick** [comment]
Actually, any token containing a `.` is affected, e.g. `:div.foo`. There's probably a whole set of punctuation marks that cause this kind of behaviour.
